### ISSUE TITLE
Flaky Analyzers on latest .NET preview

### ISSUE THREAD

**Issue by pedro-camargo11** [dotnet/roslyn#63225]
## Describe the bug
Analyzers don't get reported correctly on the latest preview of .NET.
In VS there are cases where the analyzers don't get reported at all, or get reported twice. Whenever I run `dotnet build` on powershell however, the analyzers are all reported correctly.
There were also instances where I would add white space and the analyzers would no longer be reported.

## Steps To Repro
Add the following code to a sample Minimal API Program.cs:
```C#
builder.WebHost.UseStartup(typeof(Startup));

app.MapGet("/hello/{name?}", (string name) => $"Hello {name}");

builder.Host.ConfigureLogging(logging =>
{
    logging.AddConsole();
});

for (var i = 0; i < 100; i++)
{
    _ = stackalloc byte[1];
}

public class Startup { }
```
This should fire 4 analyzers

- ASP0010
- ASP0011
- ASP0007
- CA2014

I have seen cases each analyzer is reported twice in VS, or not reported at all. They are always reported correctly on the CLI, suggesting this is a VS issue.

## .NET version
7.0.100-preview.7.22376.2

## Anything Else
Example where analyzers are reported twice:
<img width="864" alt="image" src="https://user-images.githubusercontent.com/106186580/182913366-55bd9471-de75-4a5c-b274-96d65df10c3b.png">

Example where analyzers are not reported at all:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/106186580/182913706-50dcc5d9-dd3c-43a9-8ffa-02c5d3bbe6d5.png">

**Comment by jaredpar** [comment]
Appears to be an error list issue.

**Comment by pedro-camargo-MSFT** [comment]
@jaredpar the underlining also doesn't work as expected, so I'm not sure if it's exclusively an error list issue.
![image](https://user-images.githubusercontent.com/106186580/183171795-f3920a31-8c5f-4716-a269-20bff459dc1d.png)

**Comment by jaredpar** [comment]
> the underlining also doesn't work as expected,

Can you elaborate?

**Comment by pedro-camargo-MSFT** [comment]
> > the underlining also doesn't work as expected,
> 
> Can you elaborate?

There are no green/red "squiggles" underlining the "incorrect" code.

**Comment by jaredpar** [comment]
Gotcha. That is related to the error list. Essentially if one is out of sync then it's expected the other would be too.

**Comment by mavasani** [comment]
**Update:** Investigating this further it seems there is some race condition here specific to live diagnostics in top level code. The issue doesn't repro for any code inside non-top level statements, but I am able to get it to repro quite consistently even for a regular C# console app. Will continue to investigate.

**Comment by Youssef1313** [comment]
This might be the same bug as https://github.com/dotnet/roslyn/issues/63394 ?

**Comment by mavasani** [comment]
> This might be the same bug as #63394 ?

Possibly. Is your repro also for top level code.

By the way, verified that the issue repros even on last VS2022 17.3.2 release build, so this isn't a recent regression.

**Comment by Youssef1313** [comment]
@mavasani Yes I think it was top-level.

**Comment by mavasani** [comment]
Identified the race condition and have a fix out for it: https://github.com/dotnet/roslyn/pull/63665. I am pretty surprised that we did not get reports for this earlier as the bug should have been there ever since top-level statements were introduced.
